logserver.go

```diff
@@ -125,6 +136,78 @@ func extensionToMimeType(path string) string {
 	return "application/octet-stream"
 }
 
+// Streams a dynamically created tar.gz file with the contents of the given directory
+// Will serve a partial, corrupted response if there is a error partway through the 
+// operation as we stream the response.
+// 
+// The resultant tarball will contain a single directory containing all the files
+// so it can unpack cleanly without overwriting other files.
+func serveTarball(w http.ResponseWriter, r *http.Request, dir string) error {
+	directory, err := os.Open(dir)
+	if err != nil {
+		return err
+	}
+	// "disposition filename"
+	dfilename := strings.Trim(r.URL.Path,"/")
+	dfilename = strings.Replace(dfilename, "/","_",-1)
```

Comment:
so this replaces `/api/listing/2022-02-10/184843-BZZXEGYH` with `/api/listing/2022-02-10_184843-BZZXEGYH` ? is that... useful?

Reply:
hrm, I guess that's not what it's doing. a clearer comment, with an example, would be helpful.

Reply:
The path is just the 2022-02-10 onwards; i believe the http.Handler strips the leading /api/listing for us when we route requests here.